server/routerlicious/packages/protocol-definitions/src/summary.ts

```diff
@@ -3,7 +3,7 @@
  * Licensed under the MIT License.
  */
 
-export type SummaryObject = ISummaryCommit | ISummaryTree | ISummaryBlob | ISummaryHandle;
+export type SummaryObject = ISummaryCommit | ISummaryTree | ISummaryBlob | ISummaryHandle | ISummaryAttachment;
```

Comment:
What is the difference between ISummaryBlob and ISummaryAttachment? I am wondering whether we need this new type? You have a "BlobAttach" op type and a ".blob" path in summary. Aren't those enough to distinguish this blob vs regular blobs?

Reply:
>Aren't those enough to distinguish this blob vs regular blobs?

Yes, but SPO wants it to be a new type since they don't want to parse snapshot blobs.

Reply:
But SPO will parse based on the ".blobs" path right?

Reply:
@vladsud knows better than I do, but I believe that once GC is implemented there may be attachments at different levels of the summary.

Reply:
As per my understanding, SPO just wants a known path where all blob id/sha are present. I will wait for @vladsud to clarify.

Reply:
SPO today has two types of blobs (blobs created by storage itself as part of uploading full summary, and attachment blobs), and they have two different APIs to read those. They seem to be open to merging them and if possible, we should just use single API / single representation in storage - I'll add folks once I get their github IDs (I keep forgetting them).

As for r11s, I think we should use existing paths as much as possible, and existing blob reuse seems like should already have that path (that's where summaries are incremental, and we reference blobs from prior snapshot by ID / sha). That said, I do not remember how it works (and it's not obvious from glancing quickly through the code).

As for why we might need two types - this was (if I'm not mistaken) that in a world where we need to read blobs using two different APIs (depending on type), we need to know what type of blob it is. But I think we never get an ID from storage and read it, we always use handles which clearly tell us type of blob, so this problem does not exist, no?
I.e. we can write out what ever storage-specific representation required (SHA in case of r11s), and pretty much ignore it on read (treat it as a set that we keep adding to, and write out back as "external" blobls).

Reply:
> As for why we might need two types - this was (if I'm not mistaken) that in a world where we need to read blobs using two different APIs (depending on type), we need to know what type of blob it is. But I think we never get an ID from storage and read it, we always use handles which clearly tell us type of blob, so this problem does not exist, no?

Correct! I think type can be inferred based on drivers.

Based on @wes-carlson's reply, I thought we are using a new type cause SPO wanted to know what to parse. But based on what I have heard in the design reviews, that is probably not necessary cause we have a fixed path already for snapshot and a separate type for ops as well.

My main concern are the protocol-definition changes. If it is absolutely needed, I would possibly move them somewhere else.

I am not sure why SPO has two different APIs. I don't have their github ids either. If possible, I would like to merge those two APIs as well :)

Reply:
Adding @marcmasmsft to the discussion.
We need to figure out if in SPO we can have only one type of blob (and one API for reading & writing).
Marcelo - please also take a look at 2nd payload in description of PR.

Reply:
having only one type is fine (the new blob read api can read any blob, "snapshot blobs" uploaded during a snapshot/ summary call and "external" blobs uploaded directly)

Reply:
we're still able to tell how the blob was created (through internal metadata) and, for now, we won't return "external" blobs during a get latest snapshot call; client will need to use the blob read API; snapshot blobs can be retrieved during get latest snapshot as well as through the new API.

Reply:
so, in other words, we expect clients to reference external blobs in a snapshot call by using type "blob" and id set to the blob's id (or the blob's path in a previous snapshot, in case it was already present)